helix-view/src/view.rs

```diff
@@ -38,18 +38,22 @@ impl JumpList {
         Self { jumps, current: 0 }
     }
 
-    pub fn push(&mut self, jump: Jump) {
+    pub fn push(&mut self, jump: Jump) -> usize {
+        let mut num_removed_from_front = 0;
         self.jumps.truncate(self.current);
         // don't push duplicates
         if self.jumps.back() != Some(&jump) {
-            // If the jumplist is full, drop the oldest item.
+            // If the jumplist is full, drop the oldest items.
             while self.jumps.len() >= JUMP_LIST_CAPACITY {
-                self.jumps.pop_front();
+                if self.jumps.pop_front().is_some() {
```

Comment:
This is guaranteed to be `Some` because `jumps.len()` is greater than the capacity which is `> 0`.

Reply:
Updated